File: BOJ/Kotlin/그리디 알고리즘/2812.kt
import java.io.BufferedReader
import java.io.InputStreamReader

fun main() = with(BufferedReader(InputStreamReader(System.`in`))){
    val (N, K) = readLine().split(' ').map { it.toInt() }
    val S = readLine()
    val arr = ArrayList<Char>()
    var count = K
    for (i in S){
        while (count > 0 && arr.isNotEmpty() && arr.last().toInt() < i.toInt()){
            arr.removeAt(arr.lastIndex)
            count --
        }
        arr.add(i)
    }

    for (i in count downTo 1) arr.removeAt(arr.lastIndex)

    println(arr.joinToString(""))
}

File: BOJ/Python/그리디 알고리즘/2812.py
N, K = map(int, input().split())
arr = list(input())
temp = list()
count = K

for i in range(N):
    while count > 0 and temp and temp[-1] < arr[i]:
        temp.pop()
        count -= 1
    temp.append(arr[i])

print(''.join(temp[:N-K]))

